src/Data/Vec/Functional/Algebra/Properties.agda

```diff
@@ -0,0 +1,173 @@
+open module LD {n} = LeftDefs Carrier (_≈ᴹ_ {n}) using (Congruent)
+
+------------------------------------------------------------------------
+-- Algebraic properties of _+ᴹ_ -ᴹ_ _*ₗ_
+
++ᴹ-cong : Congruent₂ (_+ᴹ_ {n})
++ᴹ-cong x≈y u≈v _ = +-cong (x≈y _) (u≈v _)
+
++ᴹ-assoc : Associative (_+ᴹ_ {n})
++ᴹ-assoc _ _ _ _ = +-assoc _ _ _
+
++ᴹ-comm : Commutative (_+ᴹ_ {n})
++ᴹ-comm _ _ _ = +-comm _ _
+
++ᴹ-identityˡ : LeftIdentity (0ᴹ {n}) _+ᴹ_
++ᴹ-identityˡ _ _ = +-identityˡ _
+
++ᴹ-identityʳ : RightIdentity (0ᴹ {n}) _+ᴹ_
++ᴹ-identityʳ _ _ = +-identityʳ _
+
++ᴹ-identity : Identity (0ᴹ {n}) _+ᴹ_
++ᴹ-identity = +ᴹ-identityˡ , +ᴹ-identityʳ
+
+-ᴹ‿cong : Congruent₁ (-ᴹ_ {n})
+-ᴹ‿cong f _ = -‿cong (f _)
+
+-ᴹ‿inverseˡ : AD'.LeftInverse (0ᴹ {n}) -ᴹ_ _+ᴹ_
+-ᴹ‿inverseˡ _ _ = -‿inverseˡ _
+
+-ᴹ‿inverseʳ : AD'.RightInverse (0ᴹ {n}) -ᴹ_ _+ᴹ_
+-ᴹ‿inverseʳ _ _ = -‿inverseʳ _
+
+-ᴹ‿inverse : AD'.Inverse (0ᴹ {n}) -ᴹ_ _+ᴹ_
+-ᴹ‿inverse = -ᴹ‿inverseˡ , -ᴹ‿inverseʳ
+
+*ₗ-cong : Congruent SR._≈_ (_*ₗ_ {n})
+*ₗ-cong x≈y u≈v i = *-cong x≈y (u≈v i)
+
+*ₗ-zeroˡ : LD.LeftZero SR.0# (0ᴹ {n}) _*ₗ_
+*ₗ-zeroˡ f i = zeroˡ (f i)
+
+*ₗ-distribʳ : _*ₗ_ LD.DistributesOverʳ SR._+_ ⟶ (_+ᴹ_ {n})
+*ₗ-distribʳ _ _ _ _ = distribʳ _ _ _
+
+*ₗ-identityˡ : LD.LeftIdentity SR.1# (_*ₗ_ {n})
+*ₗ-identityˡ _ _ = *-identityˡ _
+
+*ₗ-assoc : LD.Associative SR._*_ (_*ₗ_ {n})
+*ₗ-assoc _ _ _ _ = *-assoc _ _ _
+
+*ₗ-zeroʳ : LD.RightZero (0ᴹ {n}) _*ₗ_
+*ₗ-zeroʳ _ _ = zeroʳ _
+
+*ₗ-distribˡ : _*ₗ_ LD.DistributesOverˡ (_+ᴹ_ {n})
+*ₗ-distribˡ _ _ _ _ = distribˡ _ _ _
+
+------------------------------------------------------------------------
+-- Structures
+
+isMagma : IsMagma (_+ᴹ_ {n})
+isMagma = record
+  { isEquivalence = ≋-isEquivalence _
+  ; ∙-cong = +ᴹ-cong
+  }
+
+isSemigroup : IsSemigroup (_+ᴹ_ {n})
+isSemigroup = record
+  { isMagma = isMagma
+  ; assoc = +ᴹ-assoc
+  }
+
+isMonoid : IsMonoid (_+ᴹ_ {n}) 0ᴹ
+isMonoid = record
+  { isSemigroup = isSemigroup
+  ; identity = +ᴹ-identity
+  }
+
+isCommutativeMonoid : IsCommutativeMonoid (_+ᴹ_ {n}) 0ᴹ
+isCommutativeMonoid = record
+  { isMonoid = isMonoid
+  ; comm = +ᴹ-comm
+  }
+
+isPreleftSemimodule : IsPreleftSemimodule semiring (_≈ᴹ_ {n}) _+ᴹ_ 0ᴹ _*ₗ_
+isPreleftSemimodule = record
+  { *ₗ-cong = *ₗ-cong
+  ; *ₗ-zeroˡ = *ₗ-zeroˡ
+  ; *ₗ-distribʳ = *ₗ-distribʳ
+  ; *ₗ-identityˡ = *ₗ-identityˡ
+  ; *ₗ-assoc = *ₗ-assoc
+  ; *ₗ-zeroʳ = *ₗ-zeroʳ
+  ; *ₗ-distribˡ = *ₗ-distribˡ
+  }
+
+isLeftSemimodule : IsLeftSemimodule semiring (_≈ᴹ_ {n}) _+ᴹ_ 0ᴹ _*ₗ_
+isLeftSemimodule = record
+  { +ᴹ-isCommutativeMonoid = isCommutativeMonoid
+  ; isPreleftSemimodule = isPreleftSemimodule
+  }
+
+isLeftModule : IsLeftModule ring (_≈ᴹ_ {n}) _+ᴹ_ 0ᴹ -ᴹ_ _*ₗ_
+isLeftModule = record
+  { isLeftSemimodule = isLeftSemimodule
+  ; -ᴹ‿cong = -ᴹ‿cong
+  ; -ᴹ‿inverse = -ᴹ‿inverse
+  }
+
+------------------------------------------------------------------------
+-- Bundles
+
+magma : ℕ → Magma _ _
+magma n = record
+  { isMagma = isMagma {n}
+  }
+
+semiGroup : ℕ → Semigroup _ _
```

Comment:
Should be `semigroup`, not `semiGroup`, for consistency with the rest of library

Reply:
solved